python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1190,9 +1196,13 @@ def execute_asset_backfill_iteration_inner(
     initial_candidates: Set[AssetKeyPartitionKey] = set()
     request_roots = not asset_backfill_data.requested_runs_for_target_roots
     if request_roots:
-        initial_candidates.update(
-            asset_backfill_data.get_target_root_asset_partitions(instance_queryer)
-        )
+        target_roots = asset_backfill_data.get_target_root_asset_partitions(instance_queryer)
+        # check if any roots have already been requested, this make the bfs search more efficent
```

Comment:
nit: can we mention in the comment why some roots may have been requested? i.e.
```
Because the code server may have failed while requesting roots, some roots may have already been requested. Checking here will reduce the amount of BFS work later in the iteration.
```